src/cljs/main/broadfcui/tooltip.cljs

```diff
@@ -0,0 +1,113 @@
+   (fn [{:keys [this props state locals]}]
+     (if (:show? @state)
+       (show (:id @locals) (r/find-dom-node this) (:hover-element props))
+       (remove-tooltip (:id @locals))))})
+
+(defn- find-tooltip [tooltips id]
+  (first (filter #(= id (:id %)) tooltips)))
+
+(defn- update-tooltip [tooltips id f & args]
+  (let [tip (find-tooltip tooltips id)
+        new? (if tip false true)
+        tip (if new? {:id id} tip)
+        tooltips (if new? (conj tooltips tip) tooltips)
+        updated (apply f tip args)]
+    (if updated
+      (mapv #(if (= id (:id %)) updated %) tooltips)
+      (filterv #(not= id (:id %)) tooltips))))
+
+(defn- remove-derefed [tooltips]
+  (filterv #(not (empty? (:references %))) tooltips))
+
+(r/defc Container
+  {:show
+   (fn [{:keys [state]} id trigger content]
+     (swap! state update :tooltips update-tooltip id
+            assoc :references #{:trigger} :trigger trigger :content content))
+   :addref
+   (fn [{:keys [state]} id reference]
+     (swap! state update :tooltips update-tooltip id update :references conj reference))
+   :deref
+   (fn [{:keys [state]} id reference]
+     (swap! state update :tooltips
+            (fn [tooltips]
+              (-> tooltips
+                  (update-tooltip id update :references disj reference)
+                  remove-derefed))))
+   :get-initial-state
+   (fn []
+     {:tooltips []})
+   :render
+   (fn [{:keys [this state]}]
+     (let [{:keys [tooltips]} @state]
+       [:div {}
+        (map (fn [tooltip]
+               (let [rect (.getBoundingClientRect (:trigger tooltip))]
+                 [:div {:style {:position "absolute"
+                                :top (+ (aget rect "top") (aget rect "height"))
+                                :left (str "calc(" (aget rect "left") "px - 1rem)")
+                                :opacity (if (:showing? tooltip) 1 0)
+                                :transition "opacity 0.05s ease"}
+                        :onMouseEnter #(this :addref (:id tooltip) :body)
+                        :onMouseLeave (fn [e]
+                                        ;; mouseleave occurs before mouseenter, so this delay avoids
+                                        ;; a flash if the mouse moves from this to the trigger
+                                        (js/setTimeout
+                                         #(this :deref (:id tooltip) :body)
+                                         0))}
+                  (:content tooltip)]))
+             tooltips)]))
+   :component-did-mount
+   (fn [{:keys [this]}]
+     (set! instance this))
+   :component-did-update
+   (fn [{:keys [state]}]
+     (when (some #(not (:showing? %)) (:tooltips @state))
+       (swap! state update :tooltips #(mapv (fn [tt] (assoc tt :showing? true)) %))))})
+
+(defn with-tooltip [trigger-element hover-element]
+  [ToolTip {:trigger-element trigger-element :hover-element hover-element}])
+
+(defn- with-default-hover-container [element]
```

Comment:
So, tooltips that aren't on the bottom, or that need to be wider or taller, will require new containers, correct?